File: tests/fixtures/aws-s3-ts/basic/no_imports.ts
// File without @aws-sdk/client-s3 import — must produce zero emits
// even though it happens to have a class named GetObjectCommand.

class GetObjectCommand {
  constructor(public opts: { Bucket: string; Key: string }) {}
}

const fake = {
  send(_cmd: GetObjectCommand) {
    return Promise.resolve();
  },
};

export async function local() {
  return fake.send(new GetObjectCommand({ Bucket: 'fake', Key: 'fake' }));
}

File: tests/fixtures/awsrust-s3/basic/src/no_imports.rs
// File without aws_sdk_s3 use — must produce zero emits.

pub struct FakeClient;

impl FakeClient {
    pub fn put_object(&self) -> &Self { self }
    pub fn bucket(&self, _b: &str) -> &Self { self }
    pub fn key(&self, _k: &str) -> &Self { self }
    pub async fn send(&self) -> anyhow::Result<()> { Ok(()) }
}

pub async fn local(c: &FakeClient) -> anyhow::Result<()> {
    c.put_object().bucket("nope").key("nope").send().await?;
    Ok(())
}
